src/models/release-asset.ts

```diff
@@ -0,0 +1,468 @@
+					body: { ...restParams, asset },
+				});
+			}
+
+			let size: number;
+			let normalizedParams: Omit<ReleaseAsset['Write'], 'asset'> & {
+				asset: File | null;
+			};
+
+			if (typeof asset === 'string') {
+				size = await getFileSize(asset);
+				if (size <= MINIMUM_MULTIPART_SIZE) {
+					const buffer = await readFileChunk(asset, 0, size);
+					normalizedParams = {
+						...restParams,
+						asset: new File([buffer], getFileBasename(asset), {
+							type: getMimeType(asset),
+						}),
+					};
+				}
+			} else {
+				size = asset.size;
+				if (size <= MINIMUM_MULTIPART_SIZE) {
+					normalizedParams = createParams as Omit<
+						ReleaseAsset['Write'],
+						'asset'
+					> & { asset: File };
+				}
+			}
+
+			if (size <= MINIMUM_MULTIPART_SIZE) {
+				return await pine.post({
+					resource: 'release_asset',
+					body: normalizedParams!,
+				});
+			} else {
+				return await createMultipartReleaseAsset(createParams, {
+					chunkSize,
+					parallelUploads,
+				});
+			}
+		},
+
+		async update(
+			id: ReleaseAssetId,
+			updateParams: Partial<
+				Pick<WriteReleaseAssetParams, 'asset_key' | 'asset'>
+			>,
+			{
+				chunkSize = DEFAULT_MULTIPART_CHUNK_SIZE,
+				parallelUploads = DEFAULT_MULTIPART_PARALLEL_UPLOAD,
+			}: UploadParams = {},
+		) {
+			const { id: releaseAssetId } = await exports.get(id, { $select: 'id' });
+			const { asset, ...restParams } = updateParams;
+
+			if (asset == null) {
+				await pine.patch({
+					id: releaseAssetId,
+					resource: 'release_asset',
+					body: restParams,
+				});
+				return;
+			}
+
+			let size: number;
+			let normalizedParams: Partial<
+				Pick<ReleaseAsset['Write'], 'asset_key' | 'asset'>
+			>;
+
+			if (typeof asset === 'string') {
+				size = await getFileSize(asset);
+				if (size <= MINIMUM_MULTIPART_SIZE) {
+					const buffer = await readFileChunk(asset, 0, size);
+					normalizedParams = {
+						...restParams,
+						asset: new File([buffer], getFileBasename(asset), {
+							type: getMimeType(asset),
+						}),
+					};
+				}
+			} else {
+				size = asset.size;
+				if (size <= MINIMUM_MULTIPART_SIZE) {
+					normalizedParams = updateParams as Partial<
+						Pick<ReleaseAsset['Write'], 'asset_key' | 'asset'>
+					>;
+				}
+			}
+
+			if (size <= MINIMUM_MULTIPART_SIZE) {
+				await pine.patch({
+					id: releaseAssetId,
+					resource: 'release_asset',
+					body: normalizedParams!,
+				});
+			} else {
+				await uploadMultipartReleaseAsset(releaseAssetId, asset, {
+					chunkSize,
+					parallelUploads,
+				});
+				if (restParams.asset_key != null) {
+					await pine.patch({
+						id: releaseAssetId,
+						resource: 'release_asset',
+						body: {
+							asset_key: restParams.asset_key,
+						},
+					});
+				}
+			}
+		},
+
+		async remove(id: ReleaseAssetId) {
+			await pine.delete({
+				resource: 'release_asset',
+				id: await getId(id),
+			});
+		},
+	};
+	return exports;
+};
+
+export default getReleaseAssetModel;
```

Comment:
Any reason you decided to go w/ default exports rather than named exports like in the other helpers?